Px.Utils/PxFile/Data/StreamUtilities.cs

```diff
@@ -0,0 +1,104 @@
+﻿using System.Runtime.CompilerServices;
+using System.Text;
+
+namespace PxUtils.PxFile.Data
+{
+    /// <summary>
+    /// Utility class for working with px file <see cref="Stream"/> instances.
+    /// </summary>
+    public static class StreamUtilities
+    {
+        /// <summary>
+        /// Finds the position of the first occurrence of a specified keyword in a given stream.
+        /// </summary>
+        /// <param name="stream">The stream to search in.</param>
+        /// <param name="keyword">The keyword to search for.</param>
+        /// <param name="conf">A configuration object that contains symbols used in the px file syntax.</param>
+        /// <param name="bufferSize">The size of the buffer to use when reading from the stream. Defaults to 4096.</param>
+        /// <returns>The position of the keyword in the stream if found, otherwise -1.</returns>
+        public static long FindKeywordPosition(Stream stream, string keyword, PxFileSyntaxConf conf, int bufferSize = 4096)
+        {
+            char entrySeparator = conf.Symbols.EntrySeparator;
+
+            byte[] keywordBytes = Encoding.ASCII.GetBytes(keyword + conf.Symbols.KeywordSeparator);
+            byte[] buffer = new byte[bufferSize];
+
+            long read;
+            int keywordIndex = 0;
+            bool searchMode = true;
+
+            do
+            {
+                read = stream.Read(buffer, 0, bufferSize);
+                long indexInBuffer = FindInBuffer(read, buffer, keywordBytes, entrySeparator, ref searchMode, ref keywordIndex);
```

Comment:
What if the keyword lands between two buffer reads ?

Reply:
Refactored the code so this is no longer an issue. Added a unit test to confirm that this is no longer an issue.